src/Nethermind/Nethermind.Core.Test/MCSLockTests.cs

```diff
@@ -88,27 +88,4 @@ public void LockFairnessTest()
         var expectedOrder = Enumerable.Range(0, numberOfThreads).ToList();
         CollectionAssert.AreEqual(expectedOrder, executionOrder, "Threads did not acquire lock in the order they were started.");
     }
-
-    [Test]
-    public void NonReentrantTest()
```

Comment:
why remove?

Reply:
Hard to test in simlar way. Current uses aren't reentrant, will keep eye out on future uses I guess 🤔